### PR TITLE
Fixes for RPM related functionality in 1.0.5-M2

### PR DESCRIPTION
**Pull request by fsat**
The fixes address the following issues.

**Ensure RPM preinst script continues normally when `rpmPrefix` is not declared**
Without this fix, when `rpmPrefix` is not declared, the `preinst` will terminate in error.

**Ensure application is running on the installed directory in the RPM-based distribution**
This was fixed as part of PR #653 which was accidentally removed in PR #661. Tests are added to prevent this regression from occurring.

### CODE REVIEW

**Comment by fsat** [comment]
**Manual Tests**

Scenario:
* Without `rpmPrefix` declared
* With `rpmPrefix` declared + default installation
* With `rpmPrefix` declared + relocated installation

For each scenario, ensure:
* Package is installed in the correct location
* Process is started
* Process is running within correct working directory
* Log file has the correct name
* Package prefix is stored with the correct value if `rpmPrefix` is defined

Tested on Centos7

---

I think the issues raised in this PR is mainly caused by the fact that I was overriding the `preinst` and start script and then retrofitting them manually into `sbt-native-packager` project.

I have now taken a different approach:
* All changes are now made within my branch of `sbt-native-packager`
* `sbt publish-local -Dproject.version=${LOCAL_VERSION}` is used to publish all changes on the local filesystem repo.
* The project to be packaged depends on the `LOCAL_VERSION` of `sbt-native-packager`

With this approach, I now have a better confidence around the changes I have made. Hopefully this turns out to be the case.

**Comment by fsat** [comment]
//cc @muuki88 and @huntc

File: src/main/resources/com/typesafe/sbt/packager/archetypes/java_server/rpm/preinst-template [hunk 1]
```diff
@@ -10,5 +10,12 @@ then
...
-[ -n "$RPM_INSTALL_PREFIX" ] && echo "PACKAGE_PREFIX=${RPM_INSTALL_PREFIX}" >> /etc/sysconfig/${{app_name}}
+if [ -e /etc/sysconfig/${{app_name}} ] ;
+then
+  sed -i 's/PACKAGE_PREFIX\=.*//g' /etc/sysconfig/${{app_name}}
+fi
+
+if [ -n "$RPM_INSTALL_PREFIX" ] ;
+then
+  echo "PACKAGE_PREFIX=${RPM_INSTALL_PREFIX}" >> /etc/sysconfig/${{app_name}}
+fi
```
Comment (huntc) [inline src/main/resources/com/typesafe/sbt/packager/archetypes/java_server/rpm/preinst-template]: Were these changes necessary?

**Comment by huntc** [comment]
One small comment, otherwise LGTM

File: src/main/resources/com/typesafe/sbt/packager/archetypes/java_server/rpm/preinst-template (hunk 1 above)
Comment (fsat) [inline src/main/resources/com/typesafe/sbt/packager/archetypes/java_server/rpm/preinst-template]: Yes, without these changes the `preinst` script will fail and abort should the `if` condition turns out to be false.

**Comment by muuki88** [comment]
LGTM. Thanks :)

**Comment by muuki88** [comment]
Release in `1.0.5-M3`

**Comment by huntc** [comment]
Thanks!
